### PR TITLE
compat ragas 0.2.7

### PR DESCRIPTION
**Pull request by Yunnglin**

### CODE REVIEW

File: evalscope/backend/rag_eval/ragas/tasks/build_distribution.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import asyncio
+
+from ragas.testset.synthesizers.multi_hop import MultiHopAbstractQuerySynthesizer, MultiHopSpecificQuerySynthesizer
+from ragas.testset.synthesizers.single_hop.specific import SingleHopSpecificQuerySynthesizer
+
+from .translate_prompt import translate_prompts
+
+
+def get_distribution(llm, distribution, language):
```
Comment (wangxingjun778) [inline evalscope/backend/rag_eval/ragas/tasks/build_distribution.py]: unused function for now ?

File: evalscope/backend/rag_eval/ragas/tasks/build_distribution.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    mapping = {
+        'simple': single_hop,
+        'multi_context': multi_hop_abs,
+        'reasoning': multi_hop_spec,
+    }
+
+    return [(mapping[key], distribution[key]) for key in mapping if key in distribution]
+
+
+def default_query_distribution(llm, kg, language):
```
Comment (wangxingjun778) [inline evalscope/backend/rag_eval/ragas/tasks/build_distribution.py]: add function docstring

File: evalscope/backend/rag_eval/ragas/tasks/build_transform.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+from ragas.testset.transforms.extractors.llm_based import NERExtractor, ThemesExtractor
+from ragas.testset.transforms.filters import CustomNodeFilter
+from ragas.testset.transforms.relationship_builders import CosineSimilarityBuilder, OverlapScoreBuilder
+from ragas.testset.transforms.splitters import HeadlineSplitter
+from ragas.utils import num_tokens_from_string
+
+from .translate_prompt import translate_prompts
+
+
+def get_transform(llm, embedding, language):
```
Comment (wangxingjun778) [inline evalscope/backend/rag_eval/ragas/tasks/build_transform.py]: unused function

File: evalscope/backend/rag_eval/ragas/tasks/build_transform.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        return node.type == NodeType.DOCUMENT
+
+    def filter_chunks(node):
+        return node.type == NodeType.CHUNK
+
+    bin_ranges = [(0, 100), (101, 500), (501, 100000)]
+    result = count_doc_length_bins(documents, bin_ranges)
+    result = {k: v / len(documents) for k, v in result.items()}
+
+    transforms = []
```
Comment (wangxingjun778) [inline evalscope/backend/rag_eval/ragas/tasks/build_transform.py]: redundant variable

File: evalscope/backend/rag_eval/ragas/tasks/build_transform.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    elif result['101-500'] >= 0.25:
+        summary_extractor = SummaryExtractor(llm=llm, filter_nodes=lambda node: filter_doc_with_num_tokens(node, 100))
+        summary_emb_extractor = EmbeddingExtractor(
+            embedding_model=embedding_model,
+            property_name='summary_embedding',
+            embed_property_name='summary',
+            filter_nodes=lambda node: filter_doc_with_num_tokens(node, 100),
+        )
+
+        cosine_sim_builder = CosineSimilarityBuilder(
```
Comment (wangxingjun778) [inline evalscope/backend/rag_eval/ragas/tasks/build_transform.py]: redundant variable

File: evalscope/backend/rag_eval/ragas/tasks/build_transform.py (hunk 3 above)
Comment (Yunnglin) [reply evalscope/backend/rag_eval/ragas/tasks/build_transform.py]: removed

File: evalscope/backend/rag_eval/ragas/tasks/build_distribution.py (hunk 1 above)
Comment (Yunnglin) [reply evalscope/backend/rag_eval/ragas/tasks/build_distribution.py]: removed

File: evalscope/backend/rag_eval/ragas/tasks/build_distribution.py (hunk 2 above)
Comment (Yunnglin) [reply evalscope/backend/rag_eval/ragas/tasks/build_distribution.py]: done

File: evalscope/backend/rag_eval/ragas/tasks/build_transform.py (hunk 3 above)
Comment (Yunnglin) [reply evalscope/backend/rag_eval/ragas/tasks/build_transform.py]: removed

File: evalscope/backend/rag_eval/ragas/tasks/build_transform.py (hunk 4 above)
Comment (Yunnglin) [reply evalscope/backend/rag_eval/ragas/tasks/build_transform.py]: removed

File: evalscope/backend/rag_eval/ragas/tasks/build_transform.py (hunk 5 above)
Comment (Yunnglin) [reply evalscope/backend/rag_eval/ragas/tasks/build_transform.py]: removed

**Pull request merged**